src/commons/navigationBar/subcomponents/SicpNavigationBar.tsx

```diff
@@ -122,312 +62,536 @@ const SicpNavigationBar: React.FC = () => {
+          setSearchAutoCompleteCouldShow(false);
+          window.location.href = `http://localhost:8000/sicpjs/${result}`;
+        }}
+        onMouseOver={() => setFocusedSearchResultIndex(index)}
+      >
+        {getDisplayedIndex(result) + focusResult(rewritedSearchData.idToContentMap[result])}
+      </div>
+    );
 
-  const handleUserSearchChange = (s: string) => {
-    setSearchQuery(s);
-    setSearchAutocompleteResults(sentenceAutoComplete(s, 250, textTrie));
+    const buildSearchResultsMenuWith = (children: React.ReactNode) => {
+      return (
+        <div
+          style={{
+            height: '4000%',
+            overflowY: 'auto',
+            position: 'absolute',
+            top: `0`,
+            left: '100%',
+            width: '300%',
+            backgroundColor: 'white',
+            overflow: 'auto',
+            margin: 0,
+            padding: 0
+          }}
+        >
+          {children}
+        </div>
+      );
+    };
+
+    return menu(
+      searchInput,
+      isUserSearchSubMenuHidden,
+      getUserSearchResults,
+      buildUserSearchResultsMenuEntry,
+      buildSearchResultsMenuWith
+    );
   };
 
-  const handleIndexSearchChange = (s: string) => {
-    setIndexSearchQuery(s);
-    setIndexAutocompleteResults(autoComplete(s, 250, indexTrie));
-  };
+  const indexSearchResultSubMenu = (searchInput: string) => {
+    const isIndexSearchSubMenuHidden = (searchInput: String) => {
+      return !(isSubmenuVisibleIndex === searchInput);
+    };
+
+    const getIndexSearchResults = () => {
+      return resultsIndex
+        .filter(result => result.id)
+        .sort((a, b) => {
+          if (a.hasSubindex && !b.hasSubindex) {
+            return 1;
+          }
+          if (!a.hasSubindex && b.hasSubindex) {
+            return -1;
+          }
+          return a.order.localeCompare(b.order);
+        });
+    };
 
-  const userSearch = (
-    <div className="userSearch" style={{ position: 'relative' }} key="userSearch">
-      <div className="userSearch-inner">
-        <div style={{ display: 'inline-flex' }}>
-          <InputGroup
-            placeholder="Search"
-            value={searchQuery}
-            onChange={event => handleUserSearchChange(event.target.value)}
-          />
-          <ControlButton
-            label="Text"
-            icon={IconNames.SEARCH}
-            onClick={() => handleSearchButton()}
-          />
+    const buildIndexSearchResultsMenuEntry = (result: any, index: number) => {
+      return (
+        <div
+          style={{
+            margin: '0',
+            width: '100%',
+            backgroundColor: focusedIndexSearchResultIndex !== index ? 'white' : 'grey',
+            border: '1px solid black'
+          }}
+          key={index}
+          onClick={() => {
+            setIndexAutoCompleteCouldShow(false);
+            setSearchAutoCompleteCouldShow(false);
+            window.location.href = `http://localhost:8000/sicpjs/${result.id}`;
```

Comment:
Ditto here, should not be hardcoding the localhost URL location here

Reply:
Sorry, forget to change this. Same as previous